File: src/Client/Logistics.OfficeApp/src/app/core/models/payment/update-payment-method.command.ts
import {AddressDto} from "../address.dto";
import {
  CardFundingType,
  PaymentMethodType,
  UsBankAccountHolderType,
  UsBankAccountType,
} from "./enums";

export interface UpdatePaymentMethodCommand {
  id: string;
  tenantId: string;
  type: PaymentMethodType;
  isDefault?: boolean;
  billingAddress?: AddressDto;

  // Card-specific
  cardHolderName?: string;
  cardBrand?: string;
  cardNumber?: string;
  cvv?: string;
  expMonth?: number;
  expYear?: number;
  fundingType?: CardFundingType;

  // US Bank account-specific
  accountNumber?: string;
  accountHolderName?: string;
  bankName?: string;
  routingNumber?: string;
  accountHolderType?: UsBankAccountHolderType;
  accountType?: UsBankAccountType;

  // International Bank Account
  swiftCode?: string;
}

File: src/Core/Logistics.Application/Commands/Payment/UpdatePaymentMethod/UpdatePaymentMethodCommand.cs
using Logistics.Domain.ValueObjects;
using Logistics.Shared.Models;
using Logistics.Shared.Consts;
using MediatR;

namespace Logistics.Application.Commands;

public class UpdatePaymentMethodCommand : IRequest<Result>
{
    public Guid Id { get; set; }
    public PaymentMethodType Type { get; set; }
    public Address? BillingAddress { get; set; }

    // Card-specific
    public string? CardHolderName { get; set; }
    public string? CardNumber { get; set; }
    public string? Cvc { get; set; }
    public int? ExpMonth { get; set; }
    public int? ExpYear { get; set; }

    // US Bank account-specific
    public string? AccountNumber { get; set; }
    public string? AccountHolderName { get; set; }
    public string? BankName { get; set; }
    public string? RoutingNumber { get; set; }
    public UsBankAccountHolderType AccountHolderType { get; set; }
    public UsBankAccountType? AccountType { get; set; }

    // International Bank Account
    public string? SwiftCode { get; set; }
}
